addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/handler/KNXBasicThingHandler.java

```diff
@@ -0,0 +1,682 @@
+                    if (tableaddress != null) {
+                        Thread.sleep(OPERATION_INTERVAL);
+                        elements = getBridgeHandler().readDeviceMemory(address, toUnsigned(tableaddress), 1, false,
+                                OPERATION_TIMEOUT);
+                        if (elements != null) {
+                            int numberOfElements = toUnsigned(elements);
+                            logger.debug("The KNX Actor with address {} uses {} group addresses", address,
+                                    numberOfElements - 1);
+
+                            byte[] addressData = null;
+                            while (addressData == null) {
+                                Thread.sleep(OPERATION_INTERVAL);
+                                addressData = getBridgeHandler().readDeviceMemory(address, toUnsigned(tableaddress) + 1,
+                                        2, false, OPERATION_TIMEOUT);
+                                if (addressData != null) {
+                                    IndividualAddress individualAddress = new IndividualAddress(addressData);
+                                    logger.debug(
+                                            "The KNX Actor with address {} its real reported individual address is  {}",
+                                            address, individualAddress);
+                                }
+                            }
+
+                            for (int i = 1; i < numberOfElements; i++) {
+                                addressData = null;
+                                while (addressData == null) {
+                                    Thread.sleep(OPERATION_INTERVAL);
+                                    addressData = getBridgeHandler().readDeviceMemory(address,
+                                            toUnsigned(tableaddress) + 1 + i * 2, 2, false, OPERATION_TIMEOUT);
+                                    if (addressData != null) {
+                                        GroupAddress groupAddress = new GroupAddress(addressData);
+                                        foundGroupAddresses.add(groupAddress);
+                                    }
+                                }
+                            }
+
+                            for (GroupAddress anAddress : foundGroupAddresses) {
+                                logger.debug("The KNX Actor with address {} uses Group Address {}", address, anAddress);
+                            }
+                        }
+                    } else {
+                        logger.warn("The KNX Actor with address {} does not expose a Group Address table", address);
+                    }
+
+                    Thread.sleep(OPERATION_INTERVAL);
+                    byte[] objecttableaddress = getBridgeHandler().readDeviceProperties(address, GROUPOBJECT_OBJECT,
+                            PID.TABLE_REFERENCE, 1, 1, true, OPERATION_TIMEOUT);
+
+                    if (objecttableaddress != null) {
```

Comment:
Is this whole block of code only for debuging if so can we wrap it in a `isDebugEnabled`?